File: app/src/main/java/android/sample/com/topreposgithub/RepoAdapter.java
package android.sample.com.topreposgithub;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import androidx.recyclerview.widget.RecyclerView;

public class RepoAdapter extends RecyclerView.Adapter<RepoAdapter.ViewHolder> {

    private List<Repository> mData;
    private LayoutInflater mInflater;

    private static final String TOPICS_DIVIDER = " / ";
    private static final String DEFAULT_LANGUAGE = "Unknown";

    RepoAdapter(Context context, List<Repository> data) {
        this.mInflater = LayoutInflater.from(context);
        this.mData = data;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = mInflater.inflate(R.layout.recyclerview_row, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Repository repository = mData.get(position);
        holder.mRepoName.setText(repository.getFullName());
        holder.mRepoDesc.setText(repository.getDescription());
        holder.mRepoStars.setText(String.valueOf(repository.getStars()));

        if(!repository.getLanguage().equals("null")) {
            holder.mRepoLanguage.setText(repository.getLanguage());
        } else {
            holder.mRepoLanguage.setText(DEFAULT_LANGUAGE);
        }

        String strTopic = "";
        for (String topic : repository.getTopics()) {
            strTopic += topic + TOPICS_DIVIDER;
        }

        holder.mRepoTopic.setText(strTopic);

    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView mRepoName;
        TextView mRepoDesc;
        TextView mRepoStars;
        TextView mRepoLanguage;
        TextView mRepoTopic;

        ViewHolder(View itemView) {
            super(itemView);

            mRepoName = itemView.findViewById(R.id.tvRepoName);
            mRepoDesc = itemView.findViewById(R.id.tvRepoDesc);
            mRepoStars = itemView.findViewById(R.id.tvRepoStars);
            mRepoLanguage = itemView.findViewById(R.id.tvRepoLanguage);
            mRepoTopic = itemView.findViewById(R.id.tvRepoTopics);
        }
    }
}

File: app/src/main/java/android/sample/com/topreposgithub/ui/RepoAdapter.kt
package android.sample.com.topreposgithub.ui

import android.view.LayoutInflater
import android.view.ViewGroup

import androidx.recyclerview.widget.RecyclerView

import android.sample.com.topreposgithub.databinding.RepoBinding
import android.sample.com.topreposgithub.model.Repo

class RepoAdapter(private var mDataList: List<Repo>): RecyclerView.Adapter<RepoAdapter.RepoViewHolder>() {

    private val TOPICS_DIVIDER = " / "
    private val DEFAULT_LANGUAGE = "Unknown"

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RepoViewHolder {
        val repoBinding = RepoBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return RepoViewHolder(repoBinding)
    }

    override fun onBindViewHolder(holder: RepoViewHolder, position: Int) {
        val repository = mDataList[position]
        holder.binding.repoName.text = repository.fullName
        holder.binding.repoDesc.text = repository.description
        holder.binding.repoStars.text = repository.stars.toString()

        if(repository.language != "null") {
            holder.binding.repoLanguage.text = repository.language
        } else {
            holder.binding.repoLanguage.text = DEFAULT_LANGUAGE
        }

        var strTopic = ""
        repository.topics.forEach {
            topic -> strTopic += topic + TOPICS_DIVIDER
        }

        holder.binding.repoTopics.text = strTopic

    }

    override fun getItemCount(): Int {
        return mDataList.size
    }

    inner class RepoViewHolder(val binding: RepoBinding) : RecyclerView.ViewHolder(binding.root)
}
